optuna/storages/_rdb/storage.py

```diff
@@ -192,15 +193,24 @@ def set_study_direction(self, study_id: int, direction: StudyDirection) -> None:
         session = self.scoped_session()
 
         study = models.StudyModel.find_or_raise_by_id(study_id, session)
-
-        if study.direction != StudyDirection.NOT_SET and study.direction != direction:
+        current_direction = tuple(
+            [d.direction for d in models.StudyDirectionModel.where_study(study, session)]
+        )[0]
```

Comment:
There are a few things I'm tweaking to make it easier to fix later when we do a multi-objective integration, like this.